packages/ui/components/overlays/src/OverlayController.js

```diff
@@ -828,6 +831,14 @@ export class OverlayController extends EventTarget {
       this._hideResolve = resolve;
     });
 
+    // save the current activeElement so we know if the user set focus to another element than the invoker of the dialog
+    // while the dialog was open.
+    // We need this in the _restoreFocus method to determine if we should focus this.elementToFocusAfterHide when the
+    // dialog is closed or keep focus on the element that the user deliberately gave focus
```

Comment:
What we mainly solved here was that the native `<dialog>` automatically does focus delegation on close if I'm correct?
By storing this right before hide, we know for sure that we get the activeElement that reflects the current app state (not the one that was set by the native dialog after hide)